### PR TITLE
BSP: Indicate foreign modules by adding the foreign segments to the name

### PR DESCRIPTION
**Pull request by lefou**

### CODE REVIEW

File: scalalib/src/mill/scalalib/bsp/BspModule.scala [hunk 1]
```diff
@@ -15,7 +15,7 @@ trait BspModule extends Module {
   /** Use to fill most fields of `BuildTarget`. */
   @internal
   def bspBuildTarget: BspBuildTarget = BspBuildTarget(
-    displayName = Some(millModuleSegments.render),
+    displayName = Some(s"${millOuterCtx.foreign.map(f => s"${f.render}.").mkString}${millModuleSegments.render}"),
```
Comment (lolgab) [inline scalalib/src/mill/scalalib/bsp/BspModule.scala:18]: Do we use the same name in Bloop? Should it be calculated in a shared place and used from both?
I remember having had clashes in Bloop as well.

Comment (lefou) [reply scalalib/src/mill/scalalib/bsp/BspModule.scala:18]: Looks like it's also adding a foreign prefix:

https://github.com/com-lihaoyi/mill/blob/c4bd33965579d718d5d854d6a9b92fa9fe3afe88/contrib/bloop/src/mill/contrib/bloop/BloopImpl.scala#L152

Comment (lefou) [reply scalalib/src/mill/scalalib/bsp/BspModule.scala]: This was also used for the project file in Bloop, but not the project name, so I fixed it and also shared the implementation as suggested.

**Review by lolgab — APPROVED** [review]
Great changes! Thank you!

**Pull request merged**